File: api/scripts/list-tenants.py
#!/usr/bin/env python3
import os, re, sys
import psycopg2

# Load .env if present
if os.path.exists('.env'):
    with open('.env') as f:
        for line in f:
            m = re.match(r'([A-Za-z_][A-Za-z0-9_]*)=(.*)', line.strip())
            if m and m.group(1) not in os.environ:
                os.environ[m.group(1)] = m.group(2)

url = os.environ.get('DATABASE_URL')
if not url:
    print('DATABASE_URL not set')
    sys.exit(1)

# Parse URL manually for psycopg2 if needed
# Format: postgresql://user:pass@host:port/db
m = re.match(r'postgresql://([^:]+):([^@]+)@([^:]+):(\d+)/(\S+)', url)
if not m:
    print(f'Cannot parse DATABASE_URL: {url}')
    sys.exit(1)
user, password, host, port, db = m.groups()

conn = psycopg2.connect(dbname=db, user=user, password=password, host=host, port=port)
cur = conn.cursor()
cur.execute('SELECT id, name FROM "Tenant" ORDER BY "createdAt" ASC LIMIT 50;')
rows = cur.fetchall()
print('Tenants:')
for r in rows:
    print(f'{r[0]} | {r[1]}')
cur.close()
conn.close()

File: api/scripts/list-tenants.js
#!/usr/bin/env node
const fs = require('fs');
// Load .env manually if DATABASE_URL not present
if (!process.env.DATABASE_URL && fs.existsSync('.env')) {
  const envText = fs.readFileSync('.env', 'utf8');
  envText.split(/\r?\n/).forEach(line => {
    const m = line.match(/^([A-Za-z_][A-Za-z0-9_]*)=(.*)$/);
    if (m) {
      const key = m[1];
      let val = m[2];
      // Strip surrounding quotes if present
      if ((val.startsWith('"') && val.endsWith('"')) || (val.startsWith('\'') && val.endsWith('\''))) {
        val = val.slice(1, -1);
      }
      process.env[key] = val;
    }
  });
}
if (!process.env.DATABASE_URL) {
  console.error('DATABASE_URL not set; cannot init Prisma');
  process.exit(1);
}
const { PrismaClient } = require('@prisma/client');
const { PrismaPg } = require('@prisma/adapter-pg');
const { Pool } = require('pg');
const pool = new Pool({ connectionString: process.env.DATABASE_URL });
const adapter = new PrismaPg(pool);
const prisma = new PrismaClient({ adapter, log: ['error'] });
(async () => {
  try {
    const tenants = await prisma.tenant.findMany({ select: { id: true, name: true }, orderBy: { createdAt: 'asc' } });
    console.log('Tenants:');
    tenants.forEach(t => console.log(`${t.id} | ${t.name}`));
  } catch (e) {
    console.error('Error listing tenants', e);
  } finally {
    await prisma.$disconnect();
    await pool.end();
  }
})();
